website/source/docs/connect/proxies/envoy.md

```diff
@@ -14,257 +14,309 @@ optionally exposing a gRPC service on the local agent that serves [Envoy's xDS
+    security benefits of Connect's mTLS and works for any TCP-based protocol.
+    Load-balancing and metrics are available at the connection level.
+  - `http` - This specified the service speaks HTTP/1.x. Envoy will setup an
+    `http_connection_manager` and will be able to load-balance requests
+    individually to available upstream services. Envoy will also emit L7 metrics
+    such as request rates broken down by HTTP response code family (2xx, 4xx, 5xx,
+    etc).
+  - `http2` - This specifies that the service speaks http2. Specifically h2c since
+    Envoy will still only connect to the local service instance via plain TCP not
+    TLS. This behaves much like `http` with L7 load-balancing and metrics but has
+    additional settings that correctly enable end-to-end http2. Envoy in the
+    future may support automatic upgrades between HTTP/1.x and http2 and not need
+    explicit configuration.
+  - `grpc` - gRPC is a common RPC protocol based on http2. In addition to the
+    http2 support above, a service with `grpc` protocol will be configured with a
+    [gRPC bridge
+    filter](https://www.envoyproxy.io/docs/envoy/v1.9.1/configuration/http_filters/grpc_http1_bridge_filter#config-http-filters-grpc-bridge)
+    that allows HTTP/1.1 calls to be translated into gRPC as well as instrumenting
+    metrics with `gRPC-status` trailer codes.
+- `local_connect_timeout_ms` - The number of milliseconds to allow when making
+  connections the local application instance before timing out. Defaults to 5000
+  (5 seconds).
+
+The following configuration items may be overridden directly in the
+`proxy.upstreams[].config` field of a [proxy service
+definition](/docs/connect/proxies.html#proxy-service-definitions) or
+[`sidecar_service`](/docs/connect/proxies/sidecar-service.html) block.
+
+- `protocol` - Same as above in main config but affects the listener setup for
+  the upstream.
+- `connect_timeout_ms` - The number of milliseconds to allow when making upstream
+  connections before timing out. Defaults to 5000
+  (5 seconds).
+
+## Advanced Configuration
+
+To support more flexibility when configuring Envoy, several options exist that
+are "lower level" and require a good knowledge of Envoy's configuration format
+and options. Many require configuring a subsection of either the bootstrap or
+dynamic configuration using your own custom protobuf config.
+
+We separate these into two sets, [Advanced Bootstrap
+Options](#advanced-bootstrap-options) and [Escape Hatch
+Overrides](#escape-hatch-overrides). Both require writing Envoy config in it's
+protobuf JSON encoding, however advanced options are smaller chunks that might
+commonly need to be set for tasks like configuring tracing, while escape hatches
+give almost complete control over the proxy setup at the expense of needing to
+manually code the entire configuration in protobuf JSON.
+
+-> **Advanced Topic!** This section covers optional ways of taking almost
```

Comment:
```suggestion
-> **Advanced Topic!** This section covers options that allow users to take almost
```

Reply:
Make yellow or red

Reply:
Yellow is the precedent set by https://www.consul.io/docs/internals/gossip.html I misremembered that being blue.

Reply:
```suggestion
~> **Advanced Topic!** This section covers options that allow users to take almost
```